sys/include/xtimer/implementation.h

```diff
@@ -104,9 +121,28 @@ static inline uint32_t _xtimer_now(void)
     } while (_xtimer_high_cnt != latched_high_cnt);
 
     return latched_high_cnt | now;
+#else
+#if (XTIMER_HZ < 1000000ul) && (STIMER_HZ >= 1000000ul)
```

Comment:
it is confusing that the conditional above is around xtimer width (XTIMER_MASK is set for timers less than 32 bits wide), while this conditional has to do with the frequency of the underlying timers.
Also, using `#elif` saves you the second `#endif` below.